driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java

```diff
@@ -131,19 +131,20 @@ public void sendFailedEvent(final Throwable t) {
         long elapsedTimeNanos = System.nanoTime() - startTimeNanos;
 
         if (loggingRequired()) {
-            String messagePrefix = "Command \"{}\" failed in {} ms";
+            String messagePrefix = "Command \"{}\" failed on database \"{}\" in {} ms";
 
             logEventMessage(messagePrefix, "Command failed", commandEventException,
                     entries -> {
                         entries.add(new Entry(COMMAND_NAME, commandName));
+                        entries.add(new Entry(DATABASE_NAME, message.getNamespace().getDatabaseName()));
                         entries.add(new Entry(DURATION_MS, elapsedTimeNanos / NANOS_PER_MILLI));
                     },
                     entries -> entries.add(new Entry(COMMAND_CONTENT, null)));
         }
 
         if (eventRequired()) {
-            sendCommandFailedEvent(message, commandName, description, elapsedTimeNanos, commandEventException, commandListener,
-                    requestContext, operationContext);
+            sendCommandFailedEvent(message, message.getNamespace().getDatabaseName(), commandName, description, elapsedTimeNanos,
```

Comment:
Nit/Optional: `message.getNamespace().getDatabaseName()` is used in several places, so it can be extracted to a variable for improved readability.